bazel/ray_deps_setup.bzl

```diff
@@ -1,128 +1,198 @@
 load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository", "new_git_repository")
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-def ray_deps_setup():
-    RULES_JVM_EXTERNAL_TAG = "2.10"
-    RULES_JVM_EXTERNAL_SHA = "1bbf2e48d07686707dd85357e9a94da775e1dbd7c464272b3664283c9c716d26"
+def github_repository(**kwargs):
+    """
+    Conveniently chooses between ZIP, git, etc. GitHub repositories.
+    We prefer ZIP downloads, since they're smaller and faster
+    due to the lack of history.
+
+    The remote parameter is required.
+    One of {commit, tag, branch} is also required.
+    The name parameter is optional but recommended.
+    Set sha256 = True if the ZIP archive hash is unknown, then update it ASAP
+    to allow caching & avoid repeated downloads on every build.
+
+    If build_file is a string, it refers to a custom build file.
+    If build_file is absent, the project is assumed to contain a build file.
+    If build_file == True, then the build file name is auto-deduced.
+
+    If path   is present , local repository is assumed.
+    If sha256 == <string>, obtains ZIP download with the given hash (fast)
+    If sha256 == True    , obtains ZIP download without knowing the hash (fast)
+    If sha256 is absent  , obtains clone via git (slow and NOT recommended)
+
+    strip_prefix is automatic by default.
+    Set strip_prefix=<string> (or False) to override.
+
+    Some common parameters (like shallow_since)
+    are automatically stripped when not applicable.
+    """
+
+    remote = kwargs.pop("remote")  # required
```

Comment:
if this is required it shouldn't be a kwarg

Reply:
We don't want it to be positional. That's not normal Bazel practice for dependency rules. It would also interfere with the canonical ordering of arguments.

Reply:
In that case can we just check that it's set and raise a useful error?

Reply:
Ok